File: CodingSamples/Java/Runtime/GenMethodTest/Program3.java
class Program {
	
	//declaring generic select method with type parameter T
	private static <T> T select(int choice, T first, T second) {
if((choice % 2) == 1)
			return first;
		return second;
	}

	//using bounded type parameter
	private static <T extends Comparable<T>> T select(T first, T second) {
		if(first.compareTo(second) > 0)
			return first;
		return second;
	}

	public static void main(String[] args) {
		if(args.length > 0){
			int s = Integer.parseInt(args[0]);
			String ss = select(s, "Monday", "Friday");
			System.out.printf("Selected String = %s%n", ss);
			double sd = select(s, 23.4, 43.2);
			System.out.printf("Selected double = %s%n", sd);
			//int si = select(s, "abcd", 1234);
		}else{
			String ss = select("Monday", "Friday");
			System.out.printf("Selected String = %s%n", ss);
			double sd = select(23.4, 43.2);
			System.out.printf("Selected double = %s%n", sd);
			Interval si = select(new Interval(5, 40), new Interval(3, 20)); 
			System.out.printf("Selected Interval = %s%n", si);
		}
	}
}

File: CodingSamples/DotNet/Runtime/GenMethodTest/DemoApp/Program3.cs
class Program
{
    //generic method with type-parameter T
    static T Select<T>(int index, T first, T second) 
    {
        if((index % 2) == 1)
            return first;
        return second;
    }

    static T Select<T>(T first, T second) where T: IComparable<T>
    {
        if(first.CompareTo(second) > 0)
            return first;
        return second;
    }

    static void Main(string[] args)
    {
        if(args.Length > 0)
        {
            int s = int.Parse(args[0]);
            string ss = Select(s, "Monday", "Friday");
            Console.WriteLine($"Selected string = {ss}");
            double sd = Select(s, 54.3, 67.8);
            Console.WriteLine($"Selected string = {sd}");
            //sd = Select(s, "abcd", 123.4);
        }
        else
        {
            string ss = Select("Monday", "Friday");
            Console.WriteLine($"Selected string = {ss}");
            double sd = Select(54.3, 67.8);
            Console.WriteLine($"Selected string = {sd}");  
            Interval si = Select(new Interval(3, 45), new Interval(2, 30));
            Console.WriteLine($"Selected Interval = {si}");  
        }
    }
}
